src/shared_modules/content_manager/src/components/CtiOffsetDownloader.hpp

```diff
@@ -0,0 +1,128 @@
+/*
+ * Wazuh content manager
+ * Copyright (C) 2015, Wazuh Inc.
+ * May 02, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _CTI_OFFSET_DOWNLOADER_HPP
+#define _CTI_OFFSET_DOWNLOADER_HPP
+
+#include "../sharedDefs.hpp"
+#include "CtiDownloader.hpp"
+#include "IURLRequest.hpp"
+#include "updaterContext.hpp"
+#include <algorithm>
+#include <string>
+
+/**
+ * @class CtiOffsetDownloader
+ *
+ * @brief Class in charge of downloading content offsets from the CTI API as a step of a chain of responsibility.
+ *
+ */
+class CtiOffsetDownloader final : public CtiDownloader
+{
+private:
+    /**
+     * @brief Download the content from the API.
+     *
+     * @param context Updater context.
+     */
+    void download(UpdaterContext& context) override
+    {
+        logDebug2(WM_CONTENTUPDATER, "CtiOffsetDownloader - Starting");
+        // Get the parameters needed to download the content.
+        getParameters(context);
+
+        // First, make a get request to the API to get the consumer offset.
+        const auto consumerLastOffset {getCtiBaseParameters(m_url).lastOffset};
+
+        // Iterate until the current offset is equal to the consumer offset.
+        while (context.currentOffset < consumerLastOffset)
+        {
+            // Calculate the offset to download
+            const auto toOffset = std::min(consumerLastOffset, context.currentOffset + 1000);
```

Comment:
shouldn't we make the value 1000 configurable?

Reply:
The amount of offsets has to be lower than 1000, due to a CTI constraint. On the other hand, adding another configuration is only done when it is strictly necessary so I don't believe that should be done.